# Final Project

**Họ và tên thành viên 1:** Trần Tuấn Đạt

**MSSV:** 18120313

**Họ và tên thành viên 2:** Bùi Trọng Hoàng

**MSSV:** 18120537

---

## Import các thư viện cần thiết

In [1]:
import pandas as ps
import numpy as np

---

## Thu thập dữ liệu

Trong bài này, ta chỉ đươn giản là dùng bộ dữ liệu đã được thu nhập sẵn. File dữ liệu được lấy từ trên [kaggle](https://www.kaggle.com/shivamb/netflix-shows). File dữ liệu được đặt tên "netflix_titles.csv".

Theo mô tả [file dữ liệu](https://www.kaggle.com/shivamb/netflix-shows):
> Netflix is one of the most popular media and video streaming platforms.
> There are over 8000 movies or tv shows available on their platform, as of mid-2021, they have over 200M Subscribers globally.
> This tabular dataset consists of listings of all the movies and tv shows available on Netflix, along with details such as - cast, directors, ratings, release year, duration, etc.

---

## Khám phá dữ liệu

### Có bao nhiêu dòng, mỗi dòng có ý nghĩa gì?

In [13]:
raw_data  = ps.read_csv('netflix_titles.csv')
n_rows = raw_data.shape[0]

print(n_rows)

8807
12


Mỗi dòng trong bộ dữ liệu sẽ chứa thông tin của một chương trình hoặc một bộ phim được đăng trên Netflix. Có tất cả là 8807 dòng dữ liệu.

### Dữ liệu có bị trùng lặp không?

In [8]:
duplicated_value = raw_data[raw_data.duplicated()]
is_duplicated = len(duplicated_value) != 0 
is_duplicated

False

### Mỗi cột có ý nghĩa gì?

In [11]:
n_cols = raw_data.shape[1]

print(n_cols)

12

Theo file mô tả dữ liệu thì dữ liệu có 12 cột:
- **Cột thứ nhất (show_id):** id của show, mỗi id là độc nhất
- **Cột thứ hai (type):** Phân loại của show
- **Cột thứ ba (title):** tựa đề của show
- **Cột thứ tư (director):** Đạo diễn của show
- **Cột thứ năm (cast):** Các diễn viên tham gia show
- **Cột thứ sáu (country):** Đất nước sản xuất show
- **Cột thứ bảy (date_added):** Ngày mà show được thêm vào Netflix
- **Cột thứ tám (release_year):** Thời gian show bắt đầu được chiếu
- **Cột thứ chín (rating):** Đánh giá của show
- **Cột thứ mười (duration):** Thời lượng của show
- **Cột thứ mười một (listed_in):** Thể loại của show
- **Cột thứ mười hai (description):** Miêu tả của show

In [14]:
raw_data.isna().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

---

## Đưa ra câu hỏi để trả lời

In [15]:
raw_data.isna().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

- Nên mời đạo diễn nào cho show truyền hình đó?
- Show truyền hình đó được sản xuất ở nước nào?

---

### Tiền xử lý

---

### Phân tích dữ liệu để trả lời